### DISCUSSION TITLE
Learning more-resolute adjacency definitions

### DISCUSSION THREAD

**Discussion by MichaelHopwood** [pyg-team/pytorch_geometric#5770] [Q&A]
I have a dense graph which ends up over-aggregating when using normal GCNs in inference; specifically, the output embeddings of many nodes (~90%) are the global average. I blame this on the adjacency definition, and will need to work on that piece separately.

I was wondering if anyone knows of some papers or procedures which derive a more-optimal adjacency definition $A^* = f(A,X)$ given also information of the node features. Additionally, could potentially use node labels (although it'd have to be semi-supervised in my case) $A^* = f(A,X,y)$.

**Comment by wsad1** [comment]
Attention based GNNs could alleviate your problem of over smoothing to an extent. [TransformerConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.TransformerConv), [GATConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GATv2Conv) etc.

I am not familiar with papers about deriving an adjacency matrix, maybe someone else can help there. But you could checkout [DynamicEdgeConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.DynamicEdgeConv) where the graph is dynamically constructed.
